network.go

```diff
@@ -286,6 +302,15 @@ func (s *sandbox) updateInterface(netHandle *netlink.Handle, iface *types.Interf
 		} else {
 			resultingIfc = iface
 		}
+
+		ch := make(chan netlink.LinkUpdate)
+		done := make(chan struct{})
+		defer close(done)
+
+		if err := netlink.LinkSubscribe(ch, done); err != nil {
```

Comment:
This is making the defer function too complex to read IMO, instead I would create a function called `setupLink()` that would internally take care of:
```go
LinkSubscribe()
LinkSetUp()
expectLinkUpdate()